k9mail/src/main/java/com/fsck/k9/controller/NonQresyncExtensionHandler.java

```diff
@@ -47,15 +47,15 @@ int continueSync(MessageDownloader messageDownloader, FlagSyncHelper flagSyncHel
         Map<String, ImapMessage> remoteUidMap = new HashMap<>();
 
         int remoteMessageCount = imapFolder.getMessageCount();
-        int remoteStart = ImapSyncInteractor.getRemoteStart(localFolder, imapFolder);
+        int remoteStart = syncHelper.getRemoteStart(localFolder, imapFolder);
 
         Timber.v("SYNC: About to fetch UIDs for messages %d through %d in folder %s",
                 remoteStart, remoteMessageCount, folderName);
 
         findRemoteMessagesToDownload(localUidMap, remoteMessages, remoteUidMap, remoteStart);
 
         if (account.syncRemoteDeletions()) {
-            imapSyncInteractor.syncRemoteDeletions(findDeletedMessageUids(localUidMap, remoteUidMap));
+            imapSyncInteractor.syncRemoteDeletions(findDeletedMessageUids(localUidMap, remoteUidMap), syncHelper);
```

Comment:
this construct seems weird to me - the imap sync interactor calls to either `NonQresyncExtensionHandler` or to `QresyncExtensionHandler`, both of which then call back to `imapSyncInteractor` for syncing remote deletions. and inside that method, the imap folder and local folders are also possibly opened anew?

this is really weird control flow. I'm also not sure if the name `syncRemoteDeletions` fits for this method - doesn't it just take a list of message uids and delete them locally? how does it differ from a method that would be called `deleteLocalMessages`?

I understand that the messages that are to be deleted come from different sources in the NonQresync and Qresync control flows, but is it necessary that deletion, but does this deletion have to happen in the `continueSync` method?